### ISSUE TITLE
Popover targets should have aria-haspopup

### ISSUE THREAD

**Issue by styu** [palantir/blueprint#4794]
In order to ensure that custom controls are keyboard accessible, elements that open menus/content in popovers should communicate that structure. This can be done with `aria-haspopup`: https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaHasPopup

#### Environment

- __Package version(s)__: 3.46.0
- __Operating System__:  MacOS
- __Browser name and version__: Chrome
